src/data/nat/parity.lean

```diff
@@ -95,6 +95,9 @@ by cases mod_two_eq_zero_or_one m with h₁ h₁;
 theorem even_add' : even (m + n) ↔ (odd m ↔ odd n) :=
 by rw [even_add, even_iff_not_odd, even_iff_not_odd, not_iff_not]
 
+@[parity_simps] theorem even_add_one : even (n + 1) ↔ ¬ even n :=
```

Comment:
This is def-eq to `even_succ`, and can easily be rewritten to it using `succ_eq_add_one`, so I don't see the point in having this theorem separately.

Reply:
For `int`, we have the lemma `even_add_one`, but we don't have that name for this lemma. My thinking is it's nice to have similar lemmas named the same, it makes it easier to guess a correct lemma name.

Reply:
Do we have `even_succ` for integers? If yes, then we shouldn't have `even_add_one` on integers either. If no, then we should ditch `even_succ` on naturals in favor of `even_add_one` with the proof you gave.

Reply:
Done!

Reply:
For nats, lemmas involving `+ 1` tend to be called both `_succ` and `_add_one`, maybe more commonly `_succ`. I think it depends on how much you feel the lemma is about the inductive structure of the nats vs a matter of arithmetic. This lemma makes it so `simp` can recursively evaluate whether or not any natural number is even or odd, so `even_succ` is arguably justified here.

I'm fine with the rename to `even_add_one` though.